File: appambit-testapp/src/main/java/com/appambit/testapp/utils/storage/StorableTest.java
package com.appambit.testapp.utils.storage;

import com.appambit.sdk.models.analytics.SessionData;

import java.io.Closeable;

public interface StorableTest extends Closeable {
    void putSessionData(SessionData sessionData);
    void updateLogSessionId(String sessionId);
    void updateEventSessionId(String sessionId);
    void updateAllEventsWithSessionId(String sessionId);
}

File: samples/kotlin-app/src/main/java/com/appambit/kotlinapp/utils/storage/StorableTest.kt
package com.appambit.kotlinapp.utils.storage

import com.appambit.sdk.models.analytics.SessionData
import java.io.Closeable

interface StorableTest : Closeable {
    fun putSessionData(sessionData: SessionData?)
    fun updateLogSessionId(sessionId: String?)
    fun updateEventSessionId(sessionId: String?)
}